src/main/java/edu/ie3/datamodel/io/connectors/CouchbaseConnector.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.connectors;
+
+import com.couchbase.client.core.diagnostics.PingResult;
+import com.couchbase.client.java.AsyncCollection;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.Collection;
+import com.couchbase.client.java.kv.GetResult;
+import com.couchbase.client.java.query.QueryResult;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * Implements a DataConnector for Couchbase. Couchbase is a JSON document based database. <br>
+ * Entities will be persisted in documents which are then mapped by a key.
+ */
+public class CouchbaseConnector implements DataConnector {
+
+  private static final String DEFAULT_URL = "127.0.0.1";
+  private static final String DEFAULT_BUCKET_NAME = "ie3_in";
+  private static final String DEFAULT_USERNAME = "Administrator";
+  private static final String DEFAULT_PASSWORD = "password";
```

Comment:
Does it make sense to provide default credentials here? Don't get me wrong but IMHO it doesn't really make sense because they are barely useful in a real world scenario aren't they? In addition, having a default constructor with these parameters might lead to more confusion than it helps. Therefore I would always prefer to enforce the user to provide credentials to made them obvious instead of hiding them in a default constructor ...

Reply:
btw this holds true for alle existing connectors we have so far. I think we already have some with default creds available that already confused me a few days ago :D